### PR TITLE
[EC-206] Implement eth_getTransactionReceipt

### PR DESCRIPTION
**Pull request by adamsmo**

### CODE REVIEW

File: src/main/scala/io/iohk/ethereum/crypto/ECIESCoder.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ import java.security.SecureRandom
 
 import io.iohk.ethereum.utils.ByteUtils
 import org.spongycastle.crypto.digests.{SHA1Digest, SHA256Digest}
-import org.spongycastle.crypto.engines.AESFastEngine
+import org.spongycastle.crypto.engines.AESEngine
```
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/crypto/ECIESCoder.scala]: @adamsmo why was this changed?

File: src/main/scala/io/iohk/ethereum/jsonrpc/TransactionReceiptResponse.scala [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  transactionIndex: BigInt,
+  blockNumber: BigInt,
+  blockHash: ByteString,
+  cumulativeGasUsed: BigInt,
+  gasUsed: BigInt,
+  contractAddress: Option[Address],
+  logs: Seq[TxLog]
+)
+
+//TODO extract to top level because it is used in FilterManager.scala
```
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/jsonrpc/TransactionReceiptResponse.scala]: Does this need to be addressed in this PR?

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala [hunk 3]
```diff
@@ -231,6 +236,48 @@ class EthService(
...
+  def getTransactionReceipt(req: GetTransactionReceiptRequest): ServiceResponse[GetTransactionReceiptResponse] = Future {
+    val result: Option[TransactionReceiptResponse] = for {
+      TransactionLocation(blockHash, txIndex) <- blockchain.getTransactionLocation(req.txHash)
+      Block(header, body) <- blockchain.getBlockByHash(blockHash)
+      stx <- body.transactionList.lift(txIndex)
+      receipts <- blockchain.getReceiptsByHash(blockHash)
+      receipt: Receipt <- receipts.lift(txIndex)
+    } yield {
+
+      //transaction with contract creation 0xb7b8cc9154896b25839ede4cd0c2ad193adf06489fdd9c0a9dfce05620c04ec1
```
Comment (ntallar) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala]: Minor detail: What does `0xb7b8cc9154896b25839ede4cd0c2ad193adf06489fdd9c0a9dfce05620c04ec1` represent? From the tests it seems to be the hash of a tx, maybe we should remove this comment?

File: src/main/scala/io/iohk/ethereum/crypto/ECIESCoder.scala (hunk 1 above)
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/crypto/ECIESCoder.scala]: `org.spongycastle.crypto.engines.AESFastEngine` is deprecated now

File: src/main/scala/io/iohk/ethereum/jsonrpc/TransactionReceiptResponse.scala (hunk 2 above)
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/jsonrpc/TransactionReceiptResponse.scala]: this should be addressed after https://github.com/input-output-hk/etc-client/pull/218 is merged

File: src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala (hunk 3 above)
Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/jsonrpc/EthService.scala]: it is hash of transaction used for testing this is transaction from blockchain, I would like to leave this information here, will update comment

File: src/main/scala/io/iohk/ethereum/jsonrpc/TransactionReceiptResponse.scala (hunk 2 above)
Comment (AlanVerbner) [inline src/main/scala/io/iohk/ethereum/jsonrpc/TransactionReceiptResponse.scala]: Ok, I would merge this one and create a new task, what do you think?

Comment (adamsmo) [inline src/main/scala/io/iohk/ethereum/jsonrpc/TransactionReceiptResponse.scala]: added EC-209
